src/config/settings.py

```diff
@@ -145,14 +146,38 @@
 
 # Static files (CSS, JavaScript, Images)
 # https://docs.djangoproject.com/en/4.2/howto/static-files/
-
-STATIC_URL = "static/"
-STATIC_ROOT = BASE_DIR / "static"
-
-MEDIA_URL = "/media/"
-MEDIA_ROOT = BASE_DIR / "media"
-
-STATICFILES_DIRS = [BASE_DIR / "static/bootstrap", BASE_DIR / "static/pdfs"]
+USE_MINIO = False
+
+if USE_MINIO:
+    # aws/minio settings
+    MINIO_ACCESS_KEY = os.getenv("MINIO_ROOT_USER")
+    MINIO_SECRET_KEY = os.getenv("MINIO_ROOT_PASSWORD")
+    MINIO_BUCKET_NAME = os.getenv("MINIO_BUCKET_NAME")
+    MINIO_ENDPOINT = os.getenv("MINIO_ENDPOINT")
+    AWS_ACCESS_KEY_ID = MINIO_ACCESS_KEY
+    AWS_SECRET_ACCESS_KEY = MINIO_SECRET_KEY
+    AWS_STORAGE_BUCKET_NAME = MINIO_BUCKET_NAME
+    AWS_S3_ENDPOINT_URL = MINIO_ENDPOINT
+    AWS_DEFAULT_ACL = "public-read"
+    AWS_S3_CUSTOM_DOMAIN = None
+    # s3 static settings
+    STATIC_LOCATION = "static"
+    STATIC_URL = f"{AWS_S3_ENDPOINT_URL}/{STATIC_LOCATION}/"
+    STATICFILES_STORAGE = "src.config.storage_backends.StaticStorage"
+    # s3 public media settings
+    PUBLIC_MEDIA_LOCATION = "media"
+    MEDIA_URL = f"{AWS_S3_ENDPOINT_URL}/{PUBLIC_MEDIA_LOCATION}/"
+    DEFAULT_FILE_STORAGE = "src.config.storage_backends.PublicMediaStorage"
+    # s3 private media settings
+    PRIVATE_MEDIA_LOCATION = "private"
+    PRIVATE_FILE_STORAGE = "src.config.storage_backends.PrivateMediaStorage"
```

Comment:
how can we be sure that file storage settings are mapped to proper storage backend?

Reply:
I read documentation
Great solution!

Reply:
@volodin28 
No, still, we have Image/File fields in models. How can I be sure that they are mapped correctly to either one of Private/Public storage?

Reply:
we can specify storage in the image field of model like this:
image = models.ImageField(upload_to="product_images/", storage=PublicMediaStorage)